Core/include/Acts/TrackFinder/CKFSourceLinkSelector.hpp

```diff
@@ -121,7 +121,7 @@ struct CKFSourceLinkSelector {
       std::visit(
           [&](const auto& calibrated) {
             // The measurement surface should be the same as parameter surface
-            assert(&calibrated.referenceSurface() == surface);
+            assert(&calibrated.referenceObject() == surface);
```

Comment:
Aside: does this need to be a value comparison? Should it not be sufficient if this compares pointers?

You can leave this as-is for this PR.

Reply:
What do you mean? This is a pointer comparison

Reply:
Ah yes, you are right? Why do we need to add `operator==` for volumes then?

Reply:
Right, this could be deprecated. I will check it.

Reply:
You are right, it is not needed anymore. Do you want me to remove it or should we keep it?

Reply:
Yes, I would remove the comparison operator. (I already said this in the second review comment, but Github seems to show the comments in a strange order; please ignore this if you have already done so)